### ISSUE TITLE
`qmake` generator produces two `LIBS` lines and no `RES` line

### ISSUE THREAD

**Issue by DmitrySokolov** [conan-io/conan#1230]
```py
class QmakeGenerator(Generator):
...
    @property
    def content(self):
...
# no RES here
        template = ('CONAN_INCLUDEPATH{dep_name} += {deps.include_paths}\n'
                    'CONAN_LIBS{dep_name} += {deps.lib_paths}\n'
                    'CONAN_BINDIRS{dep_name} += {deps.bin_paths}\n'
                    'CONAN_LIBS{dep_name} += {deps.libs}\n'
                    'CONAN_DEFINES{dep_name} += {deps.defines}\n'
                    'CONAN_QMAKE_CXXFLAGS{dep_name} += {deps.cppflags}\n'
                    'CONAN_QMAKE_CFLAGS{dep_name} += {deps.cflags}\n'
                    'CONAN_QMAKE_LFLAGS{dep_name} += {deps.sharedlinkflags}\n'
                    'CONAN_QMAKE_LFLAGS{dep_name} += {deps.exelinkflags}\n')
...
        output += ('\nCONFIG(conan_basic_setup) {\n'
                   '    INCLUDEPATH += $$CONAN_INCLUDEPATH\n'
=>                 '    LIBS += $$CONAN_LIBS\n'
                   '    BINDIRS += $$CONAN_BINDIRS\n'
=>                 '    LIBS += $$CONAN_LIBS\n'
                   '    DEFINES += $$CONAN_DEFINES\n'
                   '    QMAKE_CXXFLAGS += $$CONAN_QMAKE_CXXFLAGS\n'
                   '    QMAKE_CFLAGS += $$CONAN_QMAKE_CFLAGS\n'
                   '    QMAKE_LFLAGS += $$CONAN_QMAKE_LFLAGS\n'
                   '}\n')
```

**Comment by memsharded** [comment]
You are right, a CONAN_RES variable should be added, will do it for next release. If you are using qmake, a couple of quick questions:

- Is there a Qmake variable to represent "res" resources directories
- Is Qmake multi-config, so could it support both debug and release variables in the same configuration? Like defining LIBS, LIBS_DEBUG and LIBS_RELEASE

**Comment by DmitrySokolov** [comment]
There is no "res" directory in `qmake`. It uses

- `RESOURCES` for Qt `.qrc` files
- `RC_FILE`, `RES_FILE` for Windows resources (files)
- `QMAKE_INFO_PLIST` + `QMAKE_BUNDLE_DATA` for iOS

and so on. If a library has a resource you should add it, for instance, `RES_FILE += $$shell_path($$CONAN_RES_MYLIB/file.res)`.

`qmake` supports [release and debug](http://doc.qt.io/qt-5/qmake-variable-reference.html#config) in the same config. There are 2 methods.

> As an alternative to manually writing build type conditionals, some variables offer build-specific variants, for example QMAKE_LFLAGS_RELEASE in addition to the general QMAKE_LFLAGS. These should be used when available.

I am not sure if there are `LIBS_DEBUG` and `LIBS_RELEASE` vars (docs don't mention them but `qt_configure.prf` has some vars).

**Comment by memsharded** [comment]
Hi Dmitry,

I have submitted a PR for this: https://github.com/conan-io/conan/pull/1268

- The ``LIBS`` line was not repeated, it was doing two different things, adding the libs and adding the lib paths. 
- Now, I am using QMAKE_LIBDIRS to set the lib paths
- Added ``res_paths`` and ``build_paths`` to conan-qmake variables. Didn't set any "resources" qmake variable because they are intended for .rc files, but conan res_paths, are more general, for any resources
- Added preliminary support for Release & Debug multi-config packages.
- Added some unit testing.

However I have still 2 major questions, and as I don't know much about qmake, I would love some help:

- The usage of ``CONFIG(conan_basic_setup)`` to define the consumer callable to define qmake variables from conan-qmake variables. Could it be transformed to a qmake function?
- There are conan variables for debug/release for libs, libs_paths, etc, that are not translated to qmake variables. What would be the syntax for doing such? Something like: 
```
CONFIG( debug, debug|release ) {
    # debug
    QMAKE_LIBDIR += "path/to/debug/lib"
} else {
    # release
    QMAKE_LIBDIR += "path/to/release/lib"
}
```
Can I put such expression inside my ``conan_basic_setup()``?

Would love some help, and also some testing if possible before adding this to the release. Thanks very much.

**Comment by DmitrySokolov** [comment]
> The usage of CONFIG(conan_basic_setup) to define the consumer callable to define qmake variables from conan-qmake variables. Could it be transformed to a qmake function?

The definition of conan-qmake variables can be moved into a qmake function easily, but you still need to call that function to define conan-qmake variables.

```qmake
defineReplace(myVars) {
    EOL = $$escape_expand(\n)
    return(\
        MY_V1 = qwe $$EOL\
        MY_V2 = 123 $$EOL\
    )
}

eval( $$myVars() ) {
    message( -1- MY_V1=$$MY_V1 ; MY_V2=$$MY_V2)
}
message( -2- MY_V1=$$MY_V1 ; MY_V2=$$MY_V2)
```

> Can I put such expression inside my conan_basic_setup()?

[`CONFIG()`](http://doc.qt.io/qt-5/qmake-language.html#configuration-and-scopes) is the special qmake scope. You can easily define your own scope via `CONFIG`:

```qmake
CONFIG += mystuff

mystuff {
...
}
```

But it also provide the means to detect an active scope:

```qmake
# will be a debug config
CONFIG += release widgets qml debug
#                             ^~~~~

CONFIG(debug, debug|release) {
     # it's a debug config
}

CONFIG += mystuff no_mystuff

CONFIG(mystuff, mystuff|no_mystuff) {
    message(---mystuff---)
} else {
    message(---no mystuff---)
}
```

As usual scopes it can be placed inside other scopes:

```qmake
CONFIG += mystuff

mystuff {
    CONFIG(release, debug|release) {
        message(--- my release stuff ---)
    } else {
        message(--- my debug stuff ---)
    }
}
```

**Comment by memsharded** [comment]
Excellent, I have added the Debug/Release conditions to my PR: https://github.com/conan-io/conan/pull/1268/files. I think that changing the config to a function is not worth, it feels cleaner as a config.

I have tested it manually on Linux, and it seems to work great. Thanks very much for the info.

**Comment by memsharded** [comment]
I have created 2 auxiliary example repos and did a PR to the docs to add them:

- https://github.com/conan-io/docs/pull/213
- https://github.com/memsharded/hello_multi_config
- https://github.com/memsharded/qmake_example

**Comment by memsharded** [comment]
This has been released in 0.23, please upgrade, try it and report. Thanks!
